From 785186c78e3406025237c9c8827841ed7efbd2f0 Mon Sep 17 00:00:00 2001
From: kafei-cy <yaouioh@gmail.com>
Date: Sun, 7 Dec 2025 16:24:01 +0800
Subject: [PATCH 16/26] speed: Always reset the outlen when calling
 EVP_PKEY_derive

---
 apps/speed.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/apps/speed.c b/apps/speed.c
index b78f8548..42c4c439 100644
--- a/apps/speed.c
+++ b/apps/speed.c
@@ -1281,11 +1281,14 @@ static int FFDH_derive_key_loop(void *args)
     loopargs_t *tempargs = *(loopargs_t **) args;
     EVP_PKEY_CTX *ffdh_ctx = tempargs->ffdh_ctx[testnum];
     unsigned char *derived_secret = tempargs->secret_ff_a;
-    size_t outlen = MAX_FFDH_SIZE;
     int count;
 
-    for (count = 0; COND(ffdh_c[testnum][0]); count++)
+    for (count = 0; COND(ffdh_c[testnum][0]); count++) {
+        /* outlen can be overwritten with a too small value (no padding used) */
+        size_t outlen = MAX_FFDH_SIZE;
+
         EVP_PKEY_derive(ffdh_ctx, derived_secret, &outlen);
+    }
     return count;
 }
 #endif /* OPENSSL_NO_DH */
-- 
2.34.1

