From 183009e5db8a0afe54ec9bb0028f8d6b0abc42af Mon Sep 17 00:00:00 2001
From: kafei-cy <yaouioh@gmail.com>
Date: Sun, 7 Dec 2025 17:19:29 +0800
Subject: [PATCH 23/26] Add loongarch64 target [3.0]

---
 Configurations/10-main.conf | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index 6f3be711..2b1e6118 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -523,6 +523,13 @@ my %targets = (
         perlasm_scheme   => "linux64",
     },
 
+    # loongarch64 below refers to contemporary LoongArch Architecture
+    # specifications,
+    "linux64-loongarch64" => {
+        inherit_from     => [ "linux-generic64"],
+        perlasm_scheme   => "linux64",
+    },
+
     #### IA-32 targets...
     #### These two targets are a bit aged and are to be used on older Linux
     #### machines where gcc doesn't understand -m32 and -m64
-- 
2.34.1

